src/main/java/org/broadinstitute/hellbender/exceptions/UserException.java

```diff
@@ -104,51 +125,69 @@ public CannotHandleGzippedRef() {
         private static final long serialVersionUID = 0L;
 
 
-        public CouldNotCreateOutputFile(File file, String message, Exception e) {
-            super(String.format("Couldn't write file %s because %s with exception %s", file.getAbsolutePath(), message, getMessage(e)));
+        public CouldNotCreateOutputFile(final File file, final String message, final Exception e) {
+            this(file.getAbsolutePath(), message, e);
+        }
+
+        public CouldNotCreateOutputFile(final Path path, final String message, final Exception e) {
+            this(path.toString(), message, e);
+        }
+
+        public CouldNotCreateOutputFile(final String fileName, final String message, final Exception e) {
+            super(String.format("Couldn't write file %s because %s with exception %s", fileName, message, getMessage(e)));
         }
 
-        public CouldNotCreateOutputFile(File file, String message) {
-            super(String.format("Couldn't write file %s because %s", file.getAbsolutePath(), message));
+        public CouldNotCreateOutputFile(final File file, final String message) {
+            this(file.getAbsolutePath(), message);
         }
 
-        public CouldNotCreateOutputFile(String filename, String message, Exception e) {
-            super(String.format("Couldn't write file %s because %s with exception %s", filename, message, getMessage(e)));
+        public CouldNotCreateOutputFile(final Path path, final String message) {
+            this(path.toString(), message);
         }
 
-        public CouldNotCreateOutputFile(File file, Exception e) {
+        public CouldNotCreateOutputFile(final String fileName, final String message) {
+            super(String.format("Couldn't write file %s because %s", fileName, message));
+        }
+
+        public CouldNotCreateOutputFile(final File file, final Exception e) {
             super(String.format("Couldn't write file %s because exception %s", file.getAbsolutePath(), getMessage(e)));
         }
 
-        public CouldNotCreateOutputFile(String message, Exception e) {
+        public CouldNotCreateOutputFile(final Path path, final Exception e) {
+            super(String.format("Couldn't write file %s because exception %s", path.toString(), getMessage(e)));
+        }
+
+        public CouldNotCreateOutputFile(final String message, final Exception e) {
             super(message, e);
         }
     }
 
+    // todo - why we need this?
+    // todo - remove to use MalformedBAM?
```

Comment:
This `UserException` is redundant with `MalformedBAM`. I have in my backup branch a commit considering this: https://github.com/magicDGS/gatk/commit/15d7ca3ba732f0f8c3475c637d2655f045787b9b